app/encounter-builder/current-encounter.controller.js

```diff
@@ -10,12 +10,45 @@
         var vm = this;
         
         vm.encounter = encounter;
+        vm.generateAnother = generateAnother;

```

Comment:
Instead of 5 different functions, this could easily be one.

```
function generateEncounter(difficulty) {
	// Default to last
	difficulty = difficulty || lastDifficulty;

	// Set last
	lastDifficulty = difficulty;

	encounter.generateRandom(vm.filters, diff);
}
```